### PR TITLE
added input form and impersonate option

### PR DESCRIPTION
**Pull request by AliSzu**

### CODE REVIEW

File: src/api/jokesApi.ts [hunk 1]
```diff
@@ -4,15 +4,14 @@ import { jokesClient } from "./httpClient";
 import { PATHS } from "../constans/ApiPaths";
 
 export const jokesApi = {
-  fetchRandomJokes: async (): Promise<AxiosResponse<ChuckJoke>|AxiosError> => { 
+  fetchRandomJokes: async (params: string): Promise<AxiosResponse<ChuckJoke>|AxiosError> => {
```
Comment (rafalhadzicki) [inline src/api/jokesApi.ts]: you can put which params you're expecting exactly, for ex. `async (page: number, pageSize:number): Promise<AxiosResponse<ChuckJoke>|AxiosError> => {

File: src/api/jokesApi.ts [hunk 2]
```diff
@@ -4,15 +4,14 @@ import { jokesClient } from "./httpClient";
 import { PATHS } from "../constans/ApiPaths";
 
 export const jokesApi = {
-  fetchRandomJokes: async (): Promise<AxiosResponse<ChuckJoke>|AxiosError> => { 
+  fetchRandomJokes: async (params: string): Promise<AxiosResponse<ChuckJoke>|AxiosError> => { 
     try { 
-      const response = await jokesClient.get<ChuckJoke>(PATHS.RANDOM_CHUCK_JOKE_PATH);
+      const response = await jokesClient.get<ChuckJoke>(`${PATHS.RANDOM_CHUCK_JOKE_PATH}${params}`);
```
Comment (rafalhadzicki) [inline src/api/jokesApi.ts]: axios client has params support, you don't need to pass them that way. Instead you can use
` await jokesClient.get<ChuckJoke>(PATHS.RANDOM_CHUCK_JOKE_PATH,{
params:{param1, param2,etc}
});` 
I recommend checking axios docs: https://axios-http.com/docs/req_config

File: src/components/Image/Image.tsx [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
+import unkownImg from '../../assets/img/unknown.png'
 import classes from "./Image.module.scss";
 
 interface IImage {
   customName: boolean;
 }
 
 const Image = ({ customName }: IImage) => {
+  const [image, setImage] = useState(chuckImg)
+  const imgClass = customName ? 'imageUnknown' : 'imageChuck'
```
Comment (rafalhadzicki) [inline src/components/Image/Image.tsx]: use enum here

File: src/components/Image/Image.tsx [hunk 4]
```diff
@@ -1,17 +1,26 @@
...
+
+  useEffect(() => {
+    setImage(customName ? unkownImg : chuckImg)
+  }, [image])
+
```
Comment (rafalhadzicki) [inline src/components/Image/Image.tsx]: I don't think you need this useEffect and useState [image, setImage]

File: src/components/Image/Image.tsx [hunk 5]
```diff
@@ -1,17 +1,26 @@
+import { useEffect, useState } from "react";
 import chuckImg from "../../assets/img/chuck.png";
+import unkownImg from '../../assets/img/unknown.png'
```
Comment (rafalhadzicki) [inline src/components/Image/Image.tsx]: typo 😸 install vscode extension for typo checking https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

File: src/components/MainScreen/MainScreen.tsx [hunk 6]
```diff
@@ -6,50 +6,63 @@ import Image from "../Image/Image";
...
-        setJoke(axiosResponse.data)
-      }
+    const getJokes = async () => {
+      await fetchJokes("");
     };
-    getJoke();
+    getJokes();
   }, []);
 
+  const fetchJokes = async (params: string) => {
```
Comment (rafalhadzicki) [inline src/components/MainScreen/MainScreen.tsx]: change it the same way as I suggested I previous comment

File: src/components/MainScreen/MainScreen.tsx [hunk 7]
```diff
@@ -6,50 +6,63 @@ import Image from "../Image/Image";
...
-      const axiosResponse = await jokesApi.fetchRandomJokes();
-      if(axiosResponse instanceof AxiosError){
-        setIsError(true)
-        setErrorMessage(axiosResponse.message)
-      } else {
-        setIsError(false)
-        setJoke(axiosResponse.data)
-      }
+    const getJokes = async () => {
+      await fetchJokes("");
```
Comment (rafalhadzicki) [inline src/components/MainScreen/MainScreen.tsx]: params should be optional so you don't have to pass empty strings

File: src/components/MainScreen/MainScreen.tsx [hunk 8]
```diff
@@ -6,50 +6,63 @@ import Image from "../Image/Image";
...
-    setIsError(false)
-  }
+    setIsError(false);
+  };
+
+  const onFormSubmit = async (customName: string | undefined) => {
+    setCustomName(customName)
+    let params = category && `?category=${category}`;
+    const nameParams = params ? '&name=' : '?name='
+    params = customName ? params + `${nameParams}${customName}` : params + "";
```
Comment (rafalhadzicki) [inline src/components/MainScreen/MainScreen.tsx:48]: once you use axios params you won't need that

Comment (rafalhadzicki) [inline src/components/MainScreen/MainScreen.tsx]: once you use axios params you won't need that

**Review by rafalhadzicki — CHANGES_REQUESTED** [review]

File: src/components/MainScreen/MainScreen.tsx [hunk 9]
```diff
@@ -6,50 +6,63 @@ import Image from "../Image/Image";
...
     <>
-      <AlertSnackbar message={errorMessage!} isOpen={isError} onOpen={onSnackbarOpen} />
+      <AlertSnackbar
+        message={errorMessage}
+        isOpen={isError}
+        onOpen={onSnackbarOpen}
+      />
       <div className={classes.card}>
-        <Image customName={true} />
+        <Image customName={customName ? true : false} />
```
Comment (Szulerinio) [inline src/components/MainScreen/MainScreen.tsx]: you can use 
```suggestion
        <Image customName={!!customName} />
```

File: src/components/MainScreen/MainScreen.tsx [hunk 10]
```diff
@@ -6,50 +6,63 @@ import Image from "../Image/Image";
...
 import { AxiosError } from "axios";
+import InputForm from "../InputForm/InputForm";
 
 const MainScreen = () => {
   const [joke, setJoke] = useState<ChuckJoke>();
   const [errorMessage, setErrorMessage] = useState<string>();
-  const [category, setCategory] = useState('')
+  const [category, setCategory] = useState("");
   const [isError, setIsError] = useState(false);
+  const [customName, setCustomName] = useState<string>();
```
Comment (Szulerinio) [inline src/components/MainScreen/MainScreen.tsx]: doesn't typescript mark this as error? 
you type this state as string, yet pass no string as default state, and later in onFormSubmit pass 
setCustomName(customName)
where customName has type: 
customName: string | undefined

File: src/components/AlertSnackbar/AlertSnackbar.tsx [hunk 11]
```diff
@@ -1,7 +1,7 @@
 import classes from "./AlertSnackbar.module.scss";
 
 interface IAlertSnackbar {
-  message: string;
+  message: string | undefined
```
Comment (Szulerinio) [inline src/components/AlertSnackbar/AlertSnackbar.tsx]: this is perfectly ok, but if you don't know, you can use 
```suggestion
  message?: string
  ```

File: src/api/jokesApi.ts [hunk 12]
```diff
@@ -4,15 +4,19 @@ import { jokesClient } from "./httpClient";
 import { PATHS } from "../constans/ApiPaths";
 
 export const jokesApi = {
-  fetchRandomJokes: async (): Promise<AxiosResponse<ChuckJoke>|AxiosError> => { 
+  fetchRandomJokes: async (category?: string, name?: string): Promise<AxiosResponse<ChuckJoke>|AxiosError> => { 
     try { 
-      const response = await jokesClient.get<ChuckJoke>(PATHS.RANDOM_CHUCK_JOKE_PATH);
+      const response = await jokesClient.get<ChuckJoke>(`${PATHS.RANDOM_CHUCK_JOKE_PATH}`, {
```
Comment (rafalhadzicki) [inline src/api/jokesApi.ts]: don't need `$

**Review by rafalhadzicki — APPROVED** [review]

**Pull request merged**